src/components/auth/SocialMediaLogin.jsx

```diff
@@ -0,0 +1,20 @@
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+import QueryString from 'query-string';
+import { socialAuth } from '../../actions/index';
+
+export class SocialMediaLogin extends Component {
+  componentDidMount() {
+    // eslint-disable-next-line react/prop-types
+    const { socialAuth: socialAuthentication, history, location } = this.props;
+    const { token } = QueryString.parse(location.search);
```

Comment:
I like your implementation of getting the user's token from the query string here!